src/LocalFile.js

```diff
@@ -1,85 +1,100 @@
+  the option to specify the auth flag which will add or remove the Authorization HTTP header
+  sent when fetching the images. The parentNodeId field is needed so that Gatsby knows what 
+  node the newly downloaded File is connected to. Without this, Gatsby will delete the File node 
+  when you restart the develop server */
+  async downloadRemoteHostedFiles(urls, authFlag, parentNodeId) {
+    const urlsArray = convertToArray(urls);
+    const { auth, ...createRemoteArgsWithoutAuth } = this.createRemoteArgs;
 
-        sourceObject.localFiles___NODE = [];
+    try {
+      const fileNodePromises = urlsArray.map(url => {
+        const createRemoteArgs = authFlag ?
+          { url, parentNodeId, ...this.createRemoteArgs } : 
+          { url, parentNodeId, ...createRemoteArgsWithoutAuth };
 
-        await Promise.all(
-          urls.map(async url => {
-            let fileNode;
-            try {
-              fileNode = await createRemoteFileNode({
-                url,
-                parentNodeId: sourceObject.id,
-                ...this.createRemoteArgs
-              });
-            } catch (e) {
-              console.log(e);
-            }
+        return createRemoteFileNode(createRemoteArgs);
+      });
+      const fileNodes = await Promise.all(fileNodePromises);
 
-            if (fileNode) {
-              sourceObject.localFiles___NODE.push(fileNode.id);
-            }
-          })
-        );
-      })
-    );
-    return payload;
+      return this.validateFileNodes(fileNodes);
+    } catch (e) {
+        const URLStrings = urlsArray.reduce((URLString, url, i) => URLString + `URL ${i + 1}: ` + url + '\n', '');
+        console.log("\x1b[1;31m\u2715\x1b[0m We were unable to download images that stripe was pointing at\n" + URLStrings + `Error: ${e.message}\n`);
+      return null;
+    }
   }
 
-  async downloadFileNode(payload) {
-    const updatedData = payload.data.map(async file => {
-      let fileNode;
-      try {
-        fileNode = await createRemoteFileNode({
-          url: file.url,
-          parentNodeId: file.id,
-          ext: `.${file.type}`,
-          ...this.createRemoteArgs
-        });
-      } catch (e) {
-        console.log(e);
-      }
+  // File types objects have images that are hosted on Stripes servers.
+  async downloadStripeHostedFile(url, type, parentNodeId) {
+    try {
+      const fileNode = await createRemoteFileNode({
+        url,
+        ext: `.${type}`,
+        parentNodeId,
+        ...this.createRemoteArgs
+      });
 
-      if (fileNode) {
-        file.localFiles___NODE = [fileNode.id];
-      }
-      return file;
-    });
-    payload.data = await Promise.all(updatedData);
-    return payload;
+      return this.validateFileNodes(fileNode);
+    } catch (e) {
+      console.log(`\x1b[1;31m\u2715\x1b[0m We were unable to download images that stripe was hosting\nURL: ${url}\n` + `Error: ${e.message}\n`);
```

Comment:
1) I'm getting this error about 30 times over:

    > ✕ We were unable to download images that stripe was hosting
    URL: null
    Error: Failed to process remote content

    I think we should check that the url field is not null before acting on it.

2) The error message is a little misleading. Stripe hosted files could be many things other than images. While we're at it, capitalizing 'Stripe' would be good.